include/boost/math/cstdfloat/cstdfloat_cmath.hpp

```diff
@@ -879,7 +879,9 @@ namespace std
    using boost::math::cstdfloat::detail::fabs;
 
 #if !(defined(_GLIBCXX_USE_FLOAT128) && defined(__GNUC__) && (__GNUC__ >= 7))
+#if defined(__clang__) && !(!defined(__STRICT_ANSI__) && defined(_GLIBCXX_USE_FLOAT128)) // workaround for clang using libstdc++
```

Comment:
@jzmaddock I am not sure if this is correct. `std_abs.h` defines `std::abs(__float128_)` and clang throws an error for collision that GCC does not. `!(!defined(__STRICT_ANSI__) && defined(_GLIBCXX_USE_FLOAT128))` is just negated of what allows the float128 specialty in libstdc++. Thoughts?

Reply:
It looks reasonably correct - there is some inevitable impiracle guess work here that I guess we just can't avoid.  The tests are all passing though, so yes, looks good to me.